lib/core/src/sdk.rs

```diff
@@ -222,26 +231,22 @@ impl LiquidSdk {
                     update = updates_stream.recv() => match update {
                         Ok(update) => {
                             let _ = cloned.sync().await;
-                            let id = update.id();
-                            match cloned.persister.fetch_send_swap_by_id(id) {
-                                Ok(Some(_)) => {
-                                    match cloned.send_swap_state_handler.on_new_status(&update).await {
-                                        Ok(_) => info!("Successfully handled Send Swap {id} update"),
-                                        Err(e) => error!("Failed to handle Send Swap {id} update: {e}")
-                                    }
-                                }
+                            let id = &update.id;
+                            match cloned.persister.fetch_swap_by_id(id) {
+                                Ok(Swap::Send(_)) => match cloned.send_swap_state_handler.on_new_status(&update).await {
+                                    Ok(_) => info!("Successfully handled Send Swap {id} update"),
+                                    Err(e) => error!("Failed to handle Send Swap {id} update: {e}")
+                                },
+                                Ok(Swap::Receive(_)) => match cloned.receive_swap_state_handler.on_new_status(&update).await {
+                                    Ok(_) => info!("Successfully handled Receive Swap {id} update"),
+                                    Err(e) => error!("Failed to handle Receive Swap {id} update: {e}")
+                                },
+                                Ok(Swap::Chain(_)) => match cloned.chain_swap_state_handler.on_new_status(&update).await {
```

Comment:
@dangeross I think we might consider implementing two handlers (chain_send, chain_receive) instead of one handler that handles both.
The attempt to consolidate them into one handler perhaps reduce the code a bit but it makes it difficult to understand as some points in the code are trying to handle both use cases (such as claim) which I think lack some separation of concerns. Bottom line by reviewing the code one always needs to consider both cases in mind and given the fact that each case is complicated as it is it makes the review/maintaining more difficult than it could have been.
If you tried that and think otherwise I would be happy to hear some arguments.

Reply:
Makes sense from a maintainability point of view